target/linux/mediatek/image/mt7623.mk

```diff
@@ -9,22 +9,34 @@ define Build/banana-pi-sdcard
 	rm -f $@.boot
 	mkfs.fat -C $@.boot $(BOOTFS_BLOCKS)
 
-	./gen_banana_pi_img.sh emmc $@.emmc \
-		$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-preloader.bin
-
-	mkenvimage -s 0x2000 -o $(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-uboot.env $(UBOOT_TARGET)-uEnv.txt
+	if [ -r $(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-preloader.bin ]; then \
+		./gen_banana_pi_img.sh emmc $@.emmc \
+			$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-preloader.bin; \
+		mcopy -i $@.boot $@.emmc ::eMMCboot.bin; \
+	fi
+	mkenvimage -s $(UBOOT_ENVSIZE) -o $(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-uboot.env $(UBOOT_TARGET)-uEnv.txt
 	mcopy -i $@.boot $(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-uboot.env ::uboot.env
 	mcopy -i $@.boot $(IMAGE_KERNEL) ::uImage
-	mcopy -i $@.boot $@.emmc ::eMMCboot.bin
 	./gen_banana_pi_img.sh sd $@ \
 		$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-preloader.bin \
-		$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-u-boot.bin \
+		$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-u-boot*.bin \
+		$(UBOOT_OFFSET) \
 		$@.boot \
 		$(IMAGE_ROOTFS) \
 		$(CONFIG_BANANA_PI_BOOT_PARTSIZE) \
 		$(CONFIG_TARGET_ROOTFS_PARTSIZE)
 endef
 
+define Build/preloader
+	echo 'Want to copy $(STAGING_DIR_IMAGE)/$1-preloader.bin   to $@'
+	$(CP) $(STAGING_DIR_IMAGE)/$1-preloader.bin $@
+endef
+
+define Build/scatterfile
+	./gen_scatterfile.sh $(subst mt,MT,$(SUBTARGET)) "$1" \
+		$(subst -scatter.txt,,$(notdir $@)) "$(DEVICE_VENDOR) $(DEVICE_MODEL)" > $@
```

Comment:
Is there a specific reason for using only DEVICE_VENDOR and DEVICE_MODEL here manually, and not just using DEVICE_TITLE that is assembled from that + DEVICE_VARIANT?

Reply:
No reason; it's purely cosmetic anyway, to give the user a clue when they read the scatter text file (which is a list of regions given to the USB debricking tool). I can change it.

Reply:
If it's cosmetic, I'd go for the standard/appropriate variable which is DEVICE_TITLE:
https://github.com/openwrt/openwrt/blob/263f7e5bbd119ebed1f514c16f659a2e2a2b132c/include/image.mk#L360

I'd only use something different for a reason.

Reply:
Fixed; thanks.